WeakAuras/WeakAuras.lua

```diff
@@ -1512,6 +1512,30 @@ Broker_WeakAuras = LDB:NewDataObject("WeakAuras", {
   iconB = 1
 });
 
+
+do -- Archive stuff
+  local Archivist = select(2, ...).Archivist
+  function WeakAuras.OpenArchive()
+    if Archivist:IsInitialized() then
+      return Archivist
+    else
+      if not IsAddOnLoaded("WeakAurasArchive") then
+        local ok, reason = LoadAddOn("WeakAurasArchive")
+        if not ok then
+          error("Could not load WeakAuras Archive, reason: |cFFFF00" .. (reason or "UNKNOWN"))
+        end
+      end
+      Archivist:Initialize(WeakAurasArchive)
+    end
+    return Archivist
+  end
+
+  function WeakAuras.LoadFromArchive(storeType, storeID)
```

Comment:
This function is of particular interest for authors who want to store persistent data.

Reply:
I think that wouldn't be a good api to provide for auras.

We want to be able to delete the persistent data with the aura. And we every aura that wants to use persistent data to to load the WeakAurasArchivst addon.

I think we should provide something that is a bit more limited for addon authors.

Reply:
Hmm. I did want to provide authors the freedom to structure the data the way they like. A simple RawData store with a large amount of data becomes expensive to extract from the archive, so automatically providing something on `aura_env.saved` seemed a bit too limiting to me. Maybe there's something interesting we can do to add a code spot to define what store an author wants. Or maybe we just provide raw data to the author and see what people actually make before doing something elaborate.

Reply:
To clarify, there's no need to actually change anything yet. But I think we shouldn't promote this as a intended api for aura authors, as that requires imho a bit more infrastructure.